feature/git/dffml_feature_git/feature/operations.py

```diff
@@ -116,7 +116,7 @@
     outputs={"repo": git_repository_checked_out},
 )
 async def git_repo_checkout(repo: Dict[str, str], commit: str):
-    await check_output("git", "checkout", commit, cwd=repo["directory"])
+    await check_output("git", "checkout", commit, cwd=repo.directory)
     # NOTE Don't modify variables which are mearly references! This will create
     # more permutations than intended.
     checked_out = repo.copy()
```

Comment:
```
  File "/home/runner/work/dffml/dffml/feature/git/dffml_feature_git/feature/operations.py", line 122, in git_repo_checkout
    checked_out = repo.copy()
AttributeError: 'GitRepoSpec' object has no attribute 'copy'
```

The note here can be removed now that `repo` is an instance of `GitRepoSpec` which is immutable since it's a `NamedTuple`.

We should change this to:

```python
async def git_repo_checkout(repo: Dict[str, str], commit: str):
    await check_output("git", "checkout", commit, cwd=repo.directory)
    return {
        "repo": GitRepoCheckedOutSpec(
            URL=repo.URL,
            directory=repo.directory,
            commit=commit,
        )
    }
```